sql-native/src/main/java/org/ballerinalang/sql/parameterprocessor/DefaultResultParameterProcessor.java

```diff
@@ -0,0 +1,761 @@
+                for (Field internalField : internalStructFields) {
+                    int type = internalField.getFieldType().getTag();
+                    BString fieldName = fromString(internalField.getFieldName());
+                    Object value = dataArray[index];
+                    switch (type) {
+                        case TypeTags.INT_TAG:
+                            if (value instanceof BigDecimal) {
+                                struct.put(fieldName, ((BigDecimal) value).intValue());
+                            } else {
+                                struct.put(fieldName, value);
+                            }
+                            break;
+                        case TypeTags.FLOAT_TAG:
+                            if (value instanceof BigDecimal) {
+                                struct.put(fieldName, ((BigDecimal) value).doubleValue());
+                            } else {
+                                struct.put(fieldName, value);
+                            }
+                            break;
+                        case TypeTags.DECIMAL_TAG:
+                            if (value instanceof BigDecimal) {
+                                struct.put(fieldName, value);
+                            } else {
+                                struct.put(fieldName, ValueCreator.createDecimalValue((BigDecimal) value));
+                            }
+                            break;
+                        case TypeTags.STRING_TAG:
+                            struct.put(fieldName, value);
+                            break;
+                        case TypeTags.BOOLEAN_TAG:
+                            struct.put(fieldName, ((int) value) == 1);
+                            break;
+                        case TypeTags.OBJECT_TYPE_TAG:
+                        case TypeTags.RECORD_TYPE_TAG:
+                            struct.put(fieldName,
+                                    createUserDefinedType((Struct) value,
+                                            (StructureType) internalField.getFieldType()));
+                            break;
+                        default:
+                            createUserDefinedTypeSubtype(internalField, structType);
+                    }
+                    ++index;
+                }
+            }
+        } catch (SQLException e) {
+            throw new ApplicationError("Error while retrieving data to create " + structType.getName()
+                    + " record. ", e);
+        }
+        return struct;
+    }
+
+    @Override
+    protected void createUserDefinedTypeSubtype(Field internalField, StructureType structType)
+            throws ApplicationError {
+        throw new ApplicationError("Error while retrieving data for unsupported type " +
+                internalField.getFieldType().getName() + " to create "
+                + structType.getName() + " record.");
+    }
+
+    protected static Object[] validateNullable(Object[] objects) {
```

Comment:
IMO, this function is a bit confusing. If I understand correctly, it has 2 purposes:
1. Check if the `objects` array contains a `null` object.
2. Return the first non-null object.

Wouldn't it be better if we split this into 2 functions. eg:
1. `boolean containsNullObject(Object[] objects)`
2. `Object firstNonNullObject(Object[] objects)`

Reply:
Will update with a commit

Reply:
Resolved with 0bd8dba16b48436cd248d7e02ae4cc046b33a2ff